matrix/rotate_matrix.py

```diff
@@ -0,0 +1,91 @@
+# -*- coding: utf-8 -*-
+
+"""
+	In this problem, we want to rotate the matrix elements by 90, 180, 270 (counterclockwise)
+	Discussion in stackoverflow: 
+   https://stackoverflow.com/questions/42519/how-do-you-rotate-a-two-dimensional-array
+"""
+
+
+def rotate_90(matrix: [[]]):
+    """
+    rotate_90([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]])
+    [[4, 8, 12, 16], [3, 7, 11, 15], [2, 6, 10, 14], [1, 5, 9, 13]]
+    """
+    
+    transpose(matrix)
+    reverse_row(matrix)
+    
+
+def rotate_180(matrix: [[]]):
+    """
+    rotate_180([[1, 2, 3, 4], [5, 6, 7, 8], [9, 10, 11, 12], [13, 14, 15, 16]])
+    [[16, 15, 14, 13], [12, 11, 10, 9], [8, 7, 6, 5], [4, 3, 2, 1]]
+    """
+    
+    reverse_column(matrix)
+    reverse_row(matrix)
+    
+    """
+    OR
+    
+    reverse_row(matrix)
+    reverse_column(matrix)
```

Comment:
Can I do __reverse_row(reverse_column(matrix))__?  If not, that would be a nice feature to have.